pennylane/labs/resource_estimation/resource_config.py

```diff
@@ -0,0 +1,223 @@
+            print(f"Updated RX precision: {config.resource_op_precisions[ResourceRX]['precision']}")
+
+        .. code-block:: pycon
+
+            Default RX precision: 1e-09
+            Updated RX precision: 1e-05
+        """
+        self.resource_op_precisions[ResourceRX]["precision"] = precision
+        self.resource_op_precisions[ResourceCRX]["precision"] = precision
+        self.resource_op_precisions[ResourceRY]["precision"] = precision
+        self.resource_op_precisions[ResourceCRY]["precision"] = precision
+        self.resource_op_precisions[ResourceRZ]["precision"] = precision
+        self.resource_op_precisions[ResourceCRZ]["precision"] = precision
+
+    def set_decomp(
+        self,
+        op_type: type[ResourceOperator],
+        decomp_func: Callable,
+        decomp_type: DecompositionType | None = "base",
+    ) -> None:
+        """Set a custom function to override the default resource decomposition.
+
+        Args:
+            cls (Type[ResourceOperator]): the operator class whose decomposition is being overriden.
+            decomp_func (Callable): the new resource decomposition function to be set as default.
+            decomp_type (None | DecompositionType): the decomposition type to override. Options are
+                "adj", "pow", "controlled",
+                and "base". Default is "base".
+
+        .. note::
+
+            The new decomposition function, ``decomp_func``, should have the same signature as the one it replaces.
+            Specifically, the signature should match the :code:`resource_keys` of the base resource
+            operator class being overriden.
+
+        **Example**
+
+        .. code-block:: python
+
+            from pennylane.labs import resource_estimation as plre
+
+            def custom_res_decomp(**kwargs):
+                h = plre.resource_rep(plre.ResourceHadamard)
+                s = plre.resource_rep(plre.ResourceS)
+                return [plre.GateCount(h, 1), plre.GateCount(s, 2)]
+
+        .. code-block:: pycon
+
+            >>> print(plre.estimate_resources(plre.ResourceX(), gate_set={"Hadamard", "Z", "S"}))
+            --- Resources: ---
+            Total qubits: 1
+            Total gates : 4
+            Qubit breakdown:
+              clean qubits: 0, dirty qubits: 0, algorithmic qubits: 1
+            Gate breakdown:
+              {'Hadamard': 2, 'S': 2}
+            >>> config = plre.ResourceConfig()
+            >>> config.set_decomp(plre.ResourceX, custom_res_decomp)
+            >>> print(plre.estimate_resources(plre.ResourceX(), gate_set={"Hadamard", "Z", "S"}, config=config))
+            --- Resources: ---
+            Total qubits: 1
+            Total gates : 3
+            Qubit breakdown:
+              clean qubits: 0, dirty qubits: 0, algorithmic qubits: 1
+            Gate breakdown:
+              {'S': 1, 'Hadamard': 2}
+        """
+
+        if decomp_type == DecompositionType.ADJOINT:
```

Comment:
```suggestion
        decomp_type = DecompositionType(decomp_type)
        if decomp_type == DecompositionType.ADJOINT:
```
Just makes a little more user friendly.